diff --git a/dist/OpenAiLanguageModel.js b/dist/OpenAiLanguageModel.js
index a15ce263258dc55f9a01f5f36824a9cb568113fa..54bfc1f95d024348c3cb7792ca61c252e64b1903 100644
--- a/dist/OpenAiLanguageModel.js
+++ b/dist/OpenAiLanguageModel.js
@@ -1027,6 +1027,9 @@ const toChatCompletionsRequest = payload => {
     ...(payload.user !== undefined ? {
       user: payload.user
     } : undefined),
+    ...(payload.prompt_cache_key !== undefined ? {
+      prompt_cache_key: payload.prompt_cache_key
+    } : undefined),
     ...(payload.seed !== undefined ? {
       seed: payload.seed
     } : undefined),
diff --git a/src/OpenAiLanguageModel.ts b/src/OpenAiLanguageModel.ts
index f4fbc0290127bbdca4990b8ab78a1143b52a2858..25eb8400fcefc19db9bbefe544a38f7d756675c7 100644
--- a/src/OpenAiLanguageModel.ts
+++ b/src/OpenAiLanguageModel.ts
@@ -1630,6 +1630,7 @@ const toChatCompletionsRequest = (payload: CreateResponse): CreateResponseReques
     ...(payload.top_p !== undefined ? { top_p: payload.top_p } : undefined),
     ...(payload.max_output_tokens !== undefined ? { max_tokens: payload.max_output_tokens } : undefined),
     ...(payload.user !== undefined ? { user: payload.user } : undefined),
+    ...(payload.prompt_cache_key !== undefined ? { prompt_cache_key: payload.prompt_cache_key } : undefined),
     ...(payload.seed !== undefined ? { seed: payload.seed } : undefined),
     ...(payload.parallel_tool_calls !== undefined
       ? { parallel_tool_calls: payload.parallel_tool_calls }
