diff --git a/dist/BunHttpServer.js b/dist/BunHttpServer.js
index 61ef40ff8d64e232a1a39f9d2421ab45b24c6567..8c44fc34c204bc5cb4f6c91b51a611548544ea41 100644
--- a/dist/BunHttpServer.js
+++ b/dist/BunHttpServer.js
@@ -83,11 +83,14 @@ export const make = /*#__PURE__*/Effect.fnUntraced(function* (options) {
       }), middleware);
       function handler(request, server) {
         return new Promise((resolve, _reject) => {
+          const requestScope = Scope.forkUnsafe(scope, "parallel");
           const map = new Map(services.mapUnsafe);
           map.set(ServerRequest.HttpServerRequest.key, new BunServerRequest(request, resolve, removeHost(request.url), server));
-          const fiber = Fiber.runIn(Effect.runForkWith(ServiceMap.makeUnsafe(map))(httpEffect), scope);
+          const fiber = Fiber.runIn(Effect.runForkWith(ServiceMap.makeUnsafe(map))(httpEffect), requestScope);
           request.signal.addEventListener("abort", () => {
             fiber.interruptUnsafe(parent.id, Error.ClientAbort.annotation);
+            resolve(new Response(null, { status: 499 }));
+            Effect.runForkWith(services)(Scope.close(requestScope, Exit.interrupt(parent.id)));
           }, {
             once: true
           });
